### PR TITLE
[bitnami/metrics-server] Use different api version depending on the kubernetes version

### LINKED ISSUE #5906: [bitnami/metrics-server] uses deprecated apiregistration.k8s.io/v1beta1
metrics-server template for apiservice needs to be adapted for latest kubernetes versions

**Which chart**:
metrics-server

**Describe the bug**
apiregistration.k8s.io/v1beta1 APIService is deprecated in v1.19+, unavailable in v1.22+; use apiregistration.k8s.io/v1 APIService

**To Reproduce**
helm diff/sync

### PR DESCRIPTION
**Pull request by rafariossaa**
**Description of the change**

Changes de api version for `apiregistration.k8s.io` depending on the kubernetes version.

**Benefits**

**Possible drawbacks**

**Applicable issues**

  - fixes #5906

**Additional information**

**Checklist** 

- [ ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [ ] Variables are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/metrics-server/templates/metrics-api-service.yaml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 {{- if .Values.apiService.create -}}
+{{- if (semverCompare "<1.19-0" (include "common.capabilities.kubeVersion" .)) }}
```
Comment (andresbono) [inline bitnami/metrics-server/templates/metrics-api-service.yaml:2]:
```suggestion
{{- if semverCompare "<1.19-0" (include "common.capabilities.kubeVersion" .) }}
```

**Review by andresbono — COMMENTED** [review]
Could you bump the patch version of the chart?

**Review by andresbono — APPROVED** [review]
LGTM

**Comment by philicious** [comment]
@rafariossaa @andresbono while the apiVersion is fixed, there remains a cosmetical inconsistency as the `metadata.name` still reads `v1beta1.metrics.k8s.io`

**Comment by alvneiayu** [comment]
Hi @philicious 

If it is ok for you, let's continue this conversation in the issue reported from you, please. We will help you directly there.

Thanks a lot

Álvaro

**Pull request merged**